props/blaster_BC_buttons.h

```diff
@@ -0,0 +1,787 @@
+    if (DoJam()) return;
+    if (blaster_mode != MODE_AUTO && DoEmpty()) return;
+
+    switch (blaster_mode) {
+      case MODE_STUN:
+        SaberBase::DoEffect(EFFECT_STUN, 0);
+        shots_fired_++;
+        PVLOG_NORMAL << "******** STUN - Remaining shots = " << GetBulletCount() << "\n";
+        break;
+      case MODE_KILL:
+        SFX_blast.Select(-1);
+        SaberBase::DoEffect(EFFECT_FIRE, 0);
+        shots_fired_++;
+        PVLOG_NORMAL << "******** FIRE - Remaining shots = " << GetBulletCount() << "\n";
+        break;
+      case MODE_AUTO:
+        if (!CheckEmpty()) {
+          SelectAutoFirePair();
+          auto_firing_ = true;
+          SaberBase::SetLockup(LOCKUP_AUTOFIRE);
+          SaberBase::DoBeginLockup();
+        }
+        break;
+    }
+  }
+
+ void DoAutoFire() override {
+    if (!SFX_auto) return;
+    // First time through, let bgnauto finish first if we have one
+    if (SFX_bgnauto && GetWavPlayerPlaying(&SFX_bgnauto)) return;
+
+    // No bgnauto or bgnauto has finished playing, so initialize and start auto-firing
+    if (auto_firing_ && !auto_player_) {
+      auto_player_ = GetWavPlayerPlaying(&SFX_auto);
+      shots_fired_++;
+      auto_time_ = millis();
+      PVLOG_NORMAL << "******** AUTOFIRING - Remaining shots = " << GetBulletCount() << "\n";
+    }
+    // use auto.wav's length to count shots
+    if (auto_firing_ && auto_player_ && (millis() - auto_time_) > auto_sound_length) {
+      auto_sound_length = auto_player_->length() * 1000;
+      shots_fired_++;
+      if (CheckEmpty()) {
+        SaberBase::DoEndLockup();
+        SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+        auto_firing_ = false;
+        auto_player_.Free();
+        auto_time_ = millis(); // Reset for the first empty sound
+        return;
+      }
+      auto_time_ = millis();
+      PVLOG_NORMAL << "******** AUTOFIRING - Remaining shots = " << GetBulletCount() << "\n";
+    }
+
+    // This will mimic the hammer running on no bullets at the same rate it was firing.
+    // if there are different length auto.wavs, the last played's length is used.
+    if (!auto_firing_ && CheckEmpty() && trigger_is_pressed_ && (millis() - auto_time_) > auto_sound_length) {
+      SaberBase::DoEffect(EFFECT_EMPTY, 0);
+      auto_time_ = millis(); // Reset for the next empty sound
+    }
+ }
+
+  void Loop() override {
+    PropBase::Loop();
+    PollNextAction();
+    sound_library_.Poll(wav_player);
+    if (wav_player && !wav_player->isPlaying()) wav_player.Free();
+
+    if (blaster_mode == MODE_AUTO) {
+      DoAutoFire();
+    }
+  }
+
+  void Reload() override {
+    Blaster::Reload();
+    is_jammed_ = false;
+  }
+
+  void ClipOut() override {
+    Blaster::ClipOut();
+    no_clip_ = true;
+  }
+
+  void ClipIn() override {
+    Blaster::ClipIn();
+    is_jammed_ = false;
+    no_clip_ = false;
+  }
+
+  // Pull in parent's SetPreset, but turn the blaster on.
+  void SetPreset(int preset_num, bool announce) override {
```

Comment:
This seems to exactly the same thing as Blaster::SetPreset(), is there a reason for this override?

Reply:
Ah. It wasn't using the poweron sound as a determining factor before, but somebody talked me into it lately.
So now, yes it's the same.
Removed from here, and edited the base file's version to consolidate the conditions to one neater line.